src/api/functions.tsx

```diff
@@ -61,21 +73,20 @@ export function convertExpressionsToStrings(
 
   const functionsWithChannels = functionStates.map(
     ({ name, expression, channels }) => {
-      console.log(expression, functionStates);
+      const { functions: depFunctions, channels: depChannels } =
+        getDepsRecursively(expression, functionStates);
       return {
         name,
         expression: transformExpression(expression),
-        channels: [
-          ...(channels ?? []),
-          ...getChannelsRecursively(expression, functionStates),
-        ],
+        channels: [...(channels ?? []), ...depChannels],
+        functions: [name, ...depFunctions],
       };
     }
   );
 
   return {
     functions,
-    functionsWithChannels,
+    functionsWithDeps: functionsWithChannels,
```

Comment:
```suggestion
  const functionsWithDeps = functionStates.map(
    ({ name, expression, channels }) => {
      const { functions: depFunctions, channels: depChannels } =
        getDepsRecursively(expression, functionStates);
      return {
        name,
        expression: transformExpression(expression),
        channels: [...(channels ?? []), ...depChannels],
        functions: [name, ...depFunctions],
      };
    }
  );

  return {
    functions,
    functionsWithDeps,
```

Very minor comment but probably makes sense to rename this as well, also makes the return simpler as can go back to the shorthand